src/CodeGeneratorGo.cs

```diff
@@ -64,8 +65,17 @@ public override string ImplementationFileExtension
 
             // by convention the methods in the method group with an empty
             // name go into the client template so skip them here.
+            HashSet<string> ReservedFiles = new HashSet<string>();
+            ReservedFiles.Add("models");
+            ReservedFiles.Add("client");
+            ReservedFiles.Add("version");
+
             foreach (var methodGroup in codeModel.MethodGroups.Where(mg => !string.IsNullOrEmpty(mg.Name)))
             {
+                if (ReservedFiles.Contains(methodGroup.Name.Value.ToLowerInvariant()))

```

Comment:
You can get rid of the ToLowerInvariant() if you specify a comparison object in the ctor for the hash set.

Reply:
Fixed